src/components/position.js

```diff
@@ -1,8 +1,8 @@
-var coordinatesMixin = require('../utils/coordinates').componentMixin;
 var registerComponent = require('../core/component').registerComponent;
-var utils = require('../utils/');
 
-module.exports.Component = registerComponent('position', utils.extend({
+module.exports.Component = registerComponent('position', {
+  type: 'coordinate',
```

Comment:
I think that's assuming too much. Components can do many things with those three values but they should not be allocating vectors for each update. They should be creating the vector once in the init method and update it afterwards: myPosition.set(data.position.x, data.position.y, data.positionz)